src/sigal/gallery.py

```diff
@@ -286,6 +287,15 @@ def _get_markdown_metadata(self):
 
         return meta
 
+    @cached_property
+    def lat(self):
+        return self.markdown_metadata.get("lat", {})
```

Comment:
Can you change the properties to return `media.exif.gps.lat/lon` if not available in meta ? this would avoid the if/else in the template.

Reply:
But that would make it impossible to write a template that prefers the EXIF coordinates.
You might not want this, IMHO.

Reply:
people can still use directly `media.exif.gps`. But for `media.lat` I think it makes sense to avoid the manual coordinates or the exif ones.

Reply:
I hate to insist, but I feel I have to. I perceive the Python code as a considerably "deeper" layer than the templates. I expect that many users will be happy to work at the template layer but will want to avoid the Python layer. Thus, I expect that a simple, easily predictable behaviour for media.lat/lon and media.exif.gps makes it easy to write templates.

Now, if you really want to have a template without ifs, I suggest keeping meta.lat/lon as they are and adding a new set of variables that implement the fallback and a configuration option that defines preference to GPS or .md if both are present.

Reply:
And, BTW, come to think of it, one of the ifs will still be needed as it might be the case that there are no coordinates at either EXIF or .md. So we will still need an if, except without the else branch.

Reply:
I just had another idea: How about a config option that sets the order of preference between md, EXIF?
Then both ways are accessible at the template. If the option also includes a "default location" (say, 0,0 or someplace in the middle of the Pacific) then all photos are guaranteed to have a location and the template can be completely without checks. This has the advantage that photos without a location do not silently disappear but are placed at a location that the users know to mean that they have forgotten to add a location.